mewarpx/medocker/docker_shared/run_warpx.sh

```diff
@@ -119,21 +156,30 @@ do
     find . -type f -exec touch -a -c {} +
 done
 }
+
 # https://stackoverflow.com/questions/21465297/tee-stdout-and-stderr-to-separate-files-while-retaining-them-on-their-respective
-{ { bash $SCRIPTNAME; } > >( tee -a stdout.out ); } \
+bash $SCRIPTNAME > >( tee -a stdout.out ) \
                        2> >( tee -a stderr.err >&2 ) &
 
+# sleep so that there is time for the mpiexec process to launch
+sleep 2
+
 MAINID=$!
+MPIEXECID=$(pgrep -P $MAINID -f mpiexec)
+echo "main process id is $MAINID"
+echo "mpiexec process id: $MPIEXECID"
 checkpoint &
 CHECKID=$!
-wait -n $MAINID $CHECKID
+pingterminatewarning &
+PINGID=$!
+wait $MAINID
```

Comment:
```suggestion
wait -n $MAINID $CHECKID
```

I think this is still correct/necessary. `-n` means it proceeds if either exits. Normally it's `$MAINID` that exits. `checkpoint` exits if it times out, however. So I'm pretty sure with only `wait $MAINID` that timeout will kill `checkpoint` but not the full script, which is not desired behavior (eg WarpX will run indefinitely if it freezes for some reason).

Reply:
I don't understand why this happened but it looks like after this change (or the one below) the main process is now killed after the termination warning is received which makes the AWS job handler think the job completed successfully. See https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logsV2:log-groups/log-group/$252Faws$252Fbatch$252Fjob/log-events/warpx-p3-rgdevel$252Fdefault$252F7886b125435140c092377aac4c0701bb